contracts/settling_game/L08_Crypts_Resources.cairo

```diff
@@ -0,0 +1,382 @@
+    return ()
+end
+
+############
+# EXTERNAL #
+############
+
+@external
+func claim_resources{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    token_id : Uint256
+):
+    alloc_locals
+    let (caller) = get_caller_address()
+    let (controller) = MODULE_controller_address()
+
+    # # CONTRACT ADDRESSES
+
+    # EXTERNAL CONTRACTS
+    # Crypts ERC721 Token
+    let (crypts_address) = IModuleController.get_external_contract_address(
+        controller, ExternalContractIds.Crypts
+    )
+    # S_Crypts ERC721 Token
+    let (s_crypts_address) = IModuleController.get_external_contract_address(
+        controller, ExternalContractIds.S_Crypts
+    )
+    # Resources 1155 Token
+    let (resources_address) = IModuleController.get_external_contract_address(
+        controller, ExternalContractIds.Resources
+    )
+
+    # # INTERNAL CONTRACTS
+    # Crypts Logic Contract
+    let (crypts_logic_address) = IModuleController.get_module_address(
+        controller, ModuleIds.L07_Crypts
+    )
+
+    # FETCH OWNER
+    let (owner) = IERC721.ownerOf(s_crypts_address, token_id)
+
+    # ALLOW RESOURCE LOGIC ADDRESS TO CLAIM, BUT STILL RESTRICT
+    if caller != crypts_logic_address:
+        # Allwo users to claim directly
+        MODULE_ERC721_owner_check(token_id, ExternalContractIds.S_Crypts)
+        tempvar syscall_ptr = syscall_ptr
+        tempvar range_check_ptr = range_check_ptr
+        tempvar pedersen_ptr = pedersen_ptr
+    else:
+        # Or allow the Crypts contract to claim on unsettle()
+        tempvar syscall_ptr = syscall_ptr
+        tempvar range_check_ptr = range_check_ptr
+        tempvar pedersen_ptr = pedersen_ptr
+    end
+
+    let (local resource_ids : Uint256*) = alloc()
+    let (local user_resources_value : Uint256*) = alloc()  # How many of this resource get minted
+
+    # FETCH CRYPT DATA
+    let (crypts_data : CryptData) = crypts_IERC721.fetch_crypt_data(crypts_address, token_id)
+
+    # CALC DAYS
+    let (days, remainder) = days_accrued(token_id)
+
+    with_attr error_message("RESOURCES: Nothing Claimable."):
+        assert_not_zero(days)
+    end
+
+    # GET ENVIRONMENT
+    let (r_output, r_resource_id) = get_output_per_environment(crypts_data.environment)
+
+    # CHECK IF LEGENDARY
+    let r_legendary = crypts_data.legendary
+
+    # CHECK HOW MANY RESOURCES * DAYS WE SHOULD GIVE OUT
+    let (r_user_resources_value) = calculate_resource_output(
+        days, token_id, r_resource_id, r_output, r_legendary
+    )
+
+    # ADD VALUES TO TEMP ARRAY FOR EACH AVAILABLE RESOURCE
+    assert resource_ids[0] = Uint256(r_resource_id, 0)
+    assert user_resources_value[0] = r_user_resources_value
+
+    # MINT USERS RESOURCES
+    IERC1155.mintBatch(
+        resources_address,
+        owner,
+        RESOURCES_PER_CRYPT,
+        resource_ids,
+        RESOURCES_PER_CRYPT,
+        user_resources_value,
+    )
+
+    return ()
+end
+
+@external
+func upgrade_resource{
```

Comment:
I say we trim these unused functions from the code until we are going to use them

Reply:
Resolved in 148470f